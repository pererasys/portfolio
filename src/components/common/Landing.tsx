/**
 * @author Andrew Perera
 * Copyright (C) 2021
 */

import Image from "next/image";
import React from "react";

import styles from "../../styles/components/common/Landing.module.scss";

interface Props {
  title: string;
  subtitle: string;
  position?: "left" | "right" | "center";
}

const Landing: React.FC<Props> = ({ position = "center", ...props }) => {
  return (
    <section className={styles.landing}>
      <div className={[styles.header, styles[position]].join(" ")}>
        <h1>{props.title}</h1>
        <p className={styles.subtitle}>{props.subtitle}</p>
      </div>
      <div className={styles.content}>{props.children}</div>
      <Image
        src="/images/wave.svg"
        layout="fill"
        objectFit="cover"
        objectPosition="bottom"
        className={styles.wave}
      />
    </section>
  );
};

export default Landing;
