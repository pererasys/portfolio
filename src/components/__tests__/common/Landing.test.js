import { create } from "react-test-renderer";

import Landing from "../../common/Landing";

const setup = (props) => {
  const component = create(<Landing {...props} />);

  const title = component.root.findByType("h1");
  const subtitle = component.root.findByType("p");

  return {
    title,
    subtitle,
    component,
  };
};

test("should render without crashing", () => {
  const props = { title: "Test", subtitle: "test subtitle" };
  const { component } = setup(props);

  expect(component.toJSON()).toMatchSnapshot();
});

test("should render appropriately", () => {
  const props = { title: "Test", subtitle: "test subtitle" };
  const { title, subtitle } = setup(props);

  expect(title.children).toContain(props.title);
  expect(subtitle.children).toContain(props.subtitle);
});
