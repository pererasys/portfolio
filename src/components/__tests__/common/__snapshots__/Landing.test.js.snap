// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<section>
  <div
    className=" "
  >
    <h1>
      Test
    </h1>
    <p>
      test subtitle
    </p>
  </div>
  <div />
  <div
    style={
      Object {
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "left": 0,
        "margin": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <noscript>
      <img
        decoding="async"
        src="/images/wave.svg"
        srcSet="/_next/image?url=%2Fimages%2Fwave.svg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fwave.svg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fwave.svg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fwave.svg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fwave.svg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fwave.svg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fwave.svg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fwave.svg&w=3840&q=75 3840w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "cover",
            "objectPosition": "bottom",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
    <img
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": "cover",
          "objectPosition": "bottom",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </div>
</section>
`;
